core/Decode.cpp

```diff
@@ -60,20 +61,30 @@ namespace olympia
             CREATE_SPARTA_HANDLER_WITH_DATA(Decode, receiveUopQueueCredits_, uint32_t));
         in_reorder_flush_.registerConsumerHandler(
             CREATE_SPARTA_HANDLER_WITH_DATA(Decode, handleFlush_, FlushManager::FlushingCriteria));
+        in_vset_inst_.registerConsumerHandler(
+            CREATE_SPARTA_HANDLER_WITH_DATA(Decode, process_vset_, InstPtr));
 
         sparta::StartupEvent(node, CREATE_SPARTA_HANDLER(Decode, sendInitialCredits_));
     }
 
     // Send fetch the initial credit count
-    void Decode::sendInitialCredits_() { fetch_queue_credits_outp_.send(fetch_queue_.capacity()); }
+    void Decode::sendInitialCredits_()
+    {
+        fetch_queue_credits_outp_.send(fetch_queue_.capacity());
+
+        // setting MavisIDs for vsetvl and vsetivli
+        mavis_facade_ = getMavis(getContainer());
+        mavis_vsetvl_uid_ = mavis_facade_->lookupInstructionUniqueID("vsetvl");
+        mavis_vsetivli_uid_ = mavis_facade_->lookupInstructionUniqueID("vsetivli");
```

Comment:
Do we support `vsetvli`?

Reply:
Added support in [73ca9f4](https://github.com/riscv-software-src/riscv-perf-model/pull/167/commits/73ca9f4adb0bce0b8756a9b1ab3d88bf96c589bd)